pkg/apis/crd/v1alpha1/types.go

```diff
@@ -709,3 +709,152 @@ type ExternalNodeList struct {
 
 	Items []ExternalNode `json:"items,omitempty"`
 }
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+type SupportBundleCollectionList struct {
+	metav1.TypeMeta `json:",inline"`
+	// +optional
+	metav1.ListMeta `json:"metadata,omitempty"`
+
+	Items []SupportBundleCollection `json:"items"`
+}
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+type SupportBundleCollection struct {
+	metav1.TypeMeta `json:",inline"`
+	// Standard metadata of the object.
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// Specification of the desired behavior of SupportBundleCollection.
+	Spec SupportBundleSpec `json:"spec"`
+	// Most recently observed status of the SupportBundleCollection.
+	Status SupportBundleStatus `json:"status"`
+}
+
+type SupportBundleSpec struct {
+	Nodes         *BundleNodes          `json:"nodes,omitempty"`
+	ExternalNodes []BundleExternalNodes `json:"externalNodes,omitempty"`
+	// ExpirationMinutes is the requested duration of validity of the SupportBundleCollection.
+	// A SupportBundleCollection will be marked as Failed if it does not finish before expiration.
+	// Default is 60.
+	ExpirationMinutes int32 `json:"expirationMinutes"`
+	// SinceTime defines a duration newer than which the Agent logs are expected.
+	// A valid value is like: 1d, 2h, 30min.
+	SinceTime      string                        `json:"sinceTime,omitempty"`
+	FileServer     BundleFileServer              `json:"fileServer"`
+	Authentication BundleServerAuthConfiguration `json:"authentication"`
+}
+
+// BundlePhase defines the phase in which a SupportBundleCollection is.
+type BundlePhase string
+
+const (
+	// BundlePending means the SupportBundleCollection has been accepted by the system,
+	// but it has not been processed by Antrea.
+	BundlePending BundlePhase = "Pending"
+	// BundleProcessing means the SupportBundleCollection is being processed by Antrea.
+	BundleProcessing BundlePhase = "Processing"
+	// BundleProcessed means the SupportBundleCollection has been processed by all Nodes
+	// or ExternalNodes it requires.
+	BundleProcessed BundlePhase = "Processed"
+	// BundleFailed means the SupportBundleCollection failed. It can happen when any of
+	// the required Node or ExternalNode fails
+	// to process the request or gets timeout.
+	BundleFailed BundlePhase = "Failed"
```

Comment:
I still think it sounds strange the request failed, but we still have some agents upload the bundles. Probably "failed" should be only for the case the request fails at controller, or if we want also the case that no agent uploads?